assignment-client/src/entities/EntityTreeSendThread.cpp

```diff
@@ -319,42 +329,43 @@ void EntityTreeSendThread::startNewTraversal(const ViewFrustum& view, EntityTree
                     if (_entitiesInQueue.find(entity.get()) != _entitiesInQueue.end()) {
                         return;
                     }
-                    bool success = false;
-                    AACube cube = entity->getQueryAACube(success);
-                    if (success) {
-                        if (_traversal.getCurrentView().cubeIntersectsKeyhole(cube)) {
-                            // See the DiffTraversal::First case for an explanation of the "entity is too small" check
-                            float renderAccuracy = calculateRenderAccuracy(_traversal.getCurrentView().getPosition(),
-                                                                           cube,
-                                                                           _traversal.getCurrentRootSizeScale(),
-                                                                           _traversal.getCurrentLODOffset());
-
-                            // Only send entities if they are large enough to see
-                            if (renderAccuracy > 0.0f) {
-                                if (entity->getLastEdited() > startOfCompletedTraversal ||
-                                        !_traversal.getCompletedView().cubeIntersectsKeyhole(cube)) {
-                                    float priority = _conicalView.computePriority(cube);
-                                    _sendQueue.push(PrioritizedEntity(entity, priority));
-                                    _entitiesInQueue.insert(entity.get());
-                                } else {
-                                    // If this entity was skipped last time because it was too small, we still need to send it
-                                    float lastRenderAccuracy = calculateRenderAccuracy(_traversal.getCompletedView().getPosition(),
-                                                                                       cube,
-                                                                                        _traversal.getCompletedRootSizeScale(),
-                                                                                        _traversal.getCompletedLODOffset());
-
-                                    if (lastRenderAccuracy <= 0.0f) {
+                    if (_knownState.find(entity.get()) == _knownState.end() ||
+                            (_knownState.find(entity.get()) != _knownState.end() && entity->getLastEdited() > _knownState[entity.get()])) {
```

Comment:
In the conditional above: there could be as many as three distinct lookups of the target entry in `_knownState` map.  This is inefficient.  We probably should change the logic so that it only ever performs one lookup.

Reply:
Some more thoughts on this: I think we will benefit when the traversal is _**fast**_.  It might be better to keep the traversal fast and add more things to the `_sendQueue` than we'll actually send: maybe we can filter the `_sendQueue` later (if necessary) right before we actually send stuff -- not during the traversal but during the sending of packets.